lib/icinga/notification.cpp

```diff
@@ -431,7 +431,11 @@ void Notification::BeginExecuteNotification(NotificationType type, const CheckRe
 			<< "Sending " << (reminder ? "reminder " : "") << "'" << NotificationTypeToString(type) << "' notification '"
 			<< notificationName << "' for user '" << userName << "'";
 
-		Utility::QueueAsyncCallback(std::bind(&Notification::ExecuteNotificationHelper, this, type, user, cr, force, author, text));
+		// Explicitly use Notification::Ptr to keep the reference counted while the callback is active
+		Notification::Ptr notification (this);
+		Utility::QueueAsyncCallback([=]() {
```

Comment:
`[=]`?! https://www.youtube.com/watch?v=umDr0mPuyQc

Reply:
What would you do instead? Since this function may exit before the callback is executed, these have to be copied.

Reply:
I'd capture everything explicitly.

Reply:
Updated. Bonus: this now also prevents accidentally using `this` in there.